dlt/destinations/dataset/ibis_relation.py

```diff
@@ -0,0 +1,161 @@
+        sql = ibis.to_sql(self._expression, dialect=transpile_via)
+        sql = sqlglot.transpile(sql, read=transpile_via, write=target_dialect)[0]
+        return sql
+
+    @property
+    def columns_schema(self) -> TTableSchemaColumns:
+        return self.compute_columns_schema()
+
+    @columns_schema.setter
+    def columns_schema(self, new_value: TTableSchemaColumns) -> None:
+        raise NotImplementedError("columns schema in ReadableDBAPIRelation can only be computed")
+
+    def compute_columns_schema(self) -> TTableSchemaColumns:
+        """provide schema columns for the cursor, may be filtered by selected columns"""
+        # TODO: provide column lineage tracing with sqlglot lineage
+        return None
+
+    def _proxy_expression_method(self, method_name: str, *args: Any, **kwargs: Any) -> Any:
+        """Proxy method calls to the underlying ibis expression, allowing to wrap the resulting expression in a new relation"""
+
+        # Get the method from the expression
+        method = getattr(self._expression, method_name)
+
+        # unwrap args and kwargs if they are relations
+        args = tuple(
+            arg._expression if isinstance(arg, ReadableIbisRelation) else arg for arg in args
+        )
+        kwargs = {
+            k: v._expression if isinstance(v, ReadableIbisRelation) else v
+            for k, v in kwargs.items()
+        }
+
+        # casefold string params, we assume these are column names
+        args = tuple(
+            self.sql_client.capabilities.casefold_identifier(arg) if isinstance(arg, str) else arg
+            for arg in args
+        )
+        kwargs = {
+            k: self.sql_client.capabilities.casefold_identifier(v) if isinstance(v, str) else v
+            for k, v in kwargs.items()
+        }
+
+        # Call it with provided args
+        result = method(*args, **kwargs)
+
+        # If result is an ibis expression, wrap it in a new relation else return raw result
+        if isinstance(result, Expr):
+            return self.__class__(readable_dataset=self._dataset, expression=result)
+        return result
+
+    def __getattr__(self, name: str) -> Any:
+        """Wrap all callable attributes of the expression"""
+
+        attr = getattr(self._expression, name, None)
+
+        # try casefolded name for ibis columns access
+        if attr is None:
+            name = self.sql_client.capabilities.casefold_identifier(name)
+            attr = getattr(self._expression, name, None)
+
+        if attr is None:
+            raise AttributeError(f"'{self.__class__.__name__}' object has no attribute '{name}'")
+
+        if not callable(attr):
+            return attr
+        return partial(self._proxy_expression_method, name)
+
+    def __getitem__(self, columns: Union[str, Sequence[str]]) -> "ReadableIbisRelation":
+        # casefold column-names
+        columns = [self.sql_client.capabilities.casefold_identifier(col) for col in columns]
```

Comment:
this is column getter right? so you will receive ibis `Column` object(s) right? then maybe select them from the schema like we do in regular relation?

this will cover many cases already at make ibis backward compatible with base expression